theme.js

```diff
@@ -0,0 +1,42 @@
+//Add transition class DOM and remove after 500ms
+let trans = () => {
+    document.documentElement.classList.add('transition');
+        window.setTimeout(() => {
+            document.documentElement.classList.remove('transition')
+        }, 500)
+}
+
+var isClicked;
+
+//Detect if browser prefers dark or light mode
+//Set isClicked according to preference
+
+const isDarkTheme = window.matchMedia("(prefers-color-scheme: dark)")
+if (isDarkTheme.matches) {
+        document.documentElement.setAttribute('data-theme', 'dark')
+        document.getElementById("icon").src="icon_dark.png";
+        isClicked = 1;
+    } else {
+        document.documentElement.setAttribute('data-theme', 'light')
+        document.getElementById("icon").src="icon.png";
+        isClicked = 0;
+}
```

Comment:
This only matches once at page load. If you [use a listener](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Testing_media_queries#receiving_query_notifications) instead, you can change the theme every time the system setting changes.